### ISSUE TITLE
`cancelAsk` hangs on an offer no longer in the book

### ISSUE THREAD

**Issue by jkrivine** [mangrovedao/mangrove#153]
Here is a script that leads to the issue:
```js
//connecting the API to Mangrove
let mgv = await Mangrove.connect({signer:wallet});

//connecting mgv to a market
let market = await mgv.market({base:"DAI",quote:"USDC"});

// check its live
await market.consoleAsks();
await market.consoleBids();

// create a simple LP on `market`
let directLP = await mgv.liquidityProvider(market);

//Ask on market (promise base (DAI) in exchange of quote (USDC))
//LP needs to approve Mangrove for base transfer
let tx = await directLP.approveMangroveForBase();
await market.base.allowance();

// querying mangrove to know the bounty for posting a new Ask on `market`
let prov = await directLP.computeAskProvision();
tx = await directLP.fundMangrove(prov);

//Posting a new Ask
const {id:ofrId, event:askEvent} = await directLP.newAsk({wants:105, gives:104});

//taking the ask
tx = market.quote.approveMangrove();
await tx.wait();
await market.buy({wants:104,gives:105});

//retract offer hangs
await directLP.cancelAsk(ofrId,true);
```

**Comment by espendk** [comment]
@adhusson You fixed this, right?

**Comment by adhusson** [comment]
Yes, should be fixed by faf3f12bed22c
